mm/kfence/core.c

```diff
@@ -65,227 +63,22 @@ static atomic_t allocation_gate = ATOMIC_INIT(1);
 /* Wait queue to wake up heartbeat timer task. */
 static DECLARE_WAIT_QUEUE_HEAD(allocation_wait);
 
-static pgprot_t pgprot_clear_protnone_bits(pgprot_t prot)
-{
-	/*
-	 * _PAGE_GLOBAL means "global page" for present PTEs.
-	 * But, it is also used to indicate _PAGE_PROTNONE
-	 * for non-present PTEs.
-	 *
-	 * This ensures that a _PAGE_GLOBAL PTE going from
-	 * present to non-present is not confused as
-	 * _PAGE_PROTNONE.
-	 */
-	if (!(pgprot_val(prot) & _PAGE_PRESENT))
-		pgprot_val(prot) &= ~_PAGE_GLOBAL;
-
-	return prot;
-}
-
 /*
- * Some code borrowed from arch/x86/mm/pat/set_memory.c.
- * TODO(glider): need to figure out whether this code can be used on ARM64 and change it
- * accordingly.
+ * TODO(elver): With the move of arch-specific code to asm, kfence core.c got a
+ * lot simpler. Maybe we can move report.c code back here and remove the kfence/
```

Comment:
Maybe we shouldn't haste with merging the two files, because we might be adding more code in the future. Same for removing the dir.

Reply:
Yes, that's what I'm thinking. I'll keep it around for now. My preference is having it, but this depends on what MM maintainers consider best practice.